### ISSUE TITLE
Gracefully handle 127 exit code (not found) from rbenv / pyenv shims

### ISSUE THREAD

**Issue by razor-x** [neomake/neomake#537]
As discussed in #246 ([here](https://github.com/neomake/neomake/issues/246#issuecomment-237483164)), when shims from rbenv or pyenv (or possibly other *env managers) are in your path, neomake will see and call all checkers, but some might fail with exit code 127.  
Say I'm using python and have pep8 in environment A and pylint in environment B. Then there will be a shim for each in `~/.pyenv/shims`, but if I run pep8 while B is active, the command will exit with code 127 (command not found).

With the shims location in your path, Neomake "sees" both pep8 and pylint and will try to run both, but one will always "fail" with 127.

Thus, it would be nice if Neomake caught the 127 exit code and ignored / suppressed it.

**Comment by blueyed** [comment]
Agreed.  I think we should ignore all makers in the future that have returned 127 once - instead of e.g. calling them all with `--version` or `--help` to check for this.

Probably also with explicitly enabled makers (`let g:neomake_python_enabled_makers`) then?
We'll log a warning and then it is not being considered later again.

**Comment by blueyed** [comment]
Well, `fish -n` also exits with 127 in case of errors.. :/
(just tested https://github.com/neomake/neomake/pull/429)

**Comment by blueyed** [comment]
Btw: Vim-async uses 122 in case it cannot start/find a program, which Neovim reports directly.
